paddle/fluid/operators/while_op.cc

```diff
@@ -265,8 +266,10 @@ class WhileGradOpDescMaker : public framework::SingleGradOpDescMaker {
       for (auto &input_name : op->InputArgumentNames()) {
         // If the input of Op has been recorded or is generated by the forward
         // block, do not make it as input again.
+
         if (block_ins.find(input_name) != block_ins.end() ||
-            fwd_block->FindVar(input_name) != nullptr) {
+            fwd_block->FindVar(input_name) != nullptr ||
+            parent_block->FindVar(input_name) != nullptr) {
```

Comment:
Why would while_op handle this separately? All the logic has been implemented in `FindVar`, so we only need `grad_bock->FindVar(input_name) != nullptr`?

Reply:
This logic is to calculate for each operator, which inputs are the OG that calculated in outside and pass into while_grad_op by its inputs. The grad_block always contains all variables that operators needed.